### PR TITLE
New dart CLI option to create a snapshot depfile.

### PR DESCRIPTION
**Pull request by pylaligand**

### CODE REVIEW

**Review by rmacnak-google — COMMENTED** [review]
lgtm w/c

File: runtime/bin/main.cc [hunk 1]
```diff
@@ -1430,7 +1443,38 @@ bool RunMainIsolate(const char* script_name, CommandLineOptions* dart_options) {
...
 
+  if (snapshot_deps_filename != NULL) {
+    IsolateData* isolate_data =
+        reinterpret_cast<IsolateData*>(Dart_IsolateData(isolate));
+    ASSERT(isolate_data != NULL);
+    MallocGrowableArray<char*>* dependencies = isolate_data->dependencies();
+    ASSERT(dependencies != NULL);
+    File* file = File::Open(snapshot_deps_filename, File::kWriteTruncate);
+    if (file == NULL) {
+      Log::PrintErr("Error: Unable to open snapshot depfile: %s\n\n",
```
Comment (pylaligand) [inline runtime/bin/main.cc]: Done.

File: runtime/bin/main.cc [hunk 2]
```diff
@@ -1430,7 +1443,38 @@ bool RunMainIsolate(const char* script_name, CommandLineOptions* dart_options) {
...
+    for (intptr_t i = 0; i < dependencies->length(); i++) {
+      char* dep = dependencies->At(i);
+      success &= file->Print("%s ", dep);
+      free(dep);
+    }
+    success &= file->Print("\n");
+    if (!success) {
+      Log::PrintErr("Error: Unable to write dependencies file: %s\n\n",
+                    snapshot_deps_filename);
+      exit(kErrorExitCode);
```
Comment (pylaligand) [inline runtime/bin/main.cc]: Done.

File: runtime/bin/main.cc [hunk 3]
```diff
@@ -882,6 +892,9 @@ static Dart_Isolate CreateIsolateAndSetupHelper(bool is_main_isolate,
 
   IsolateData* isolate_data =
       new IsolateData(script_uri, package_root, packages_config, app_snapshot);
+  if (is_main_isolate && snapshot_deps_filename != NULL) {
```
Comment (pylaligand) [inline runtime/bin/main.cc]: Done.

**Review by rmacnak-google — APPROVED** [review]

**Pull request merged**